x-pack/plugins/actions/server/builtin_action_types/servicenow/utils.ts

```diff
@@ -134,11 +134,28 @@ export const getAxiosInstance = ({
           tokenUrl: `${snServiceUrl}/oauth_token.do`,
           connectorTokenClient,
         });
+        if (!accessToken) {
+          throw new Error(`Unable to retrieve access token for connectorId: ${connectorId}`);
+        }
         axiosConfig.headers.Authorization = accessToken;
         return axiosConfig;
       },
       (error) => {
-        Promise.reject(error);
+        return Promise.reject(error);
+      }
+    );
+    axiosInstance.interceptors.response.use(
```

Comment:
Added response interceptor to check for `4xx` errors and delete any stored access tokens for this connector. Previously, if using an access token would result in a `4xx` error, we would still continue to use it until it expired.